tests/test_search_adaptors.py

```diff
@@ -123,3 +125,162 @@ def test_vespa_search_adaptor__sensitive(fake_vespa_credentials):
+    [
+        (1, 1),
+        (1, 100),
+        (2, 1),
+        (2, 5),
+        (3, 1000),
+    ],
+)
+@pytest.mark.vespa
+def test_vespa_search_adaptor__limits(
+    fake_vespa_credentials, family_limit, max_hits_per_family
+):
+    request = SearchParameters(
+        query_string="the",
+        family_ids=[],
+        limit=family_limit,
+        max_hits_per_family=max_hits_per_family,
+    )
+    response = vespa_search(fake_vespa_credentials, request)
+
+    assert len(response.families) == family_limit
+    for fam in response.families:
+        assert len(fam.hits) <= max_hits_per_family
+
+
+@pytest.mark.vespa
+def test_vespa_search_adaptor__continuation_tokens__families(fake_vespa_credentials):
+    query_string = "the"
+    limit = 2
+    max_hits_per_family = 3
+
+    # Make an initial request to get continuation tokens and results
+    request = SearchParameters(
+        query_string=query_string,
+        limit=limit,
+        max_hits_per_family=max_hits_per_family,
+    )
+    response = vespa_search(fake_vespa_credentials, request)
+    first_family_ids = [f.id for f in response.families]
+    family_continuation = response.continuation_token
+    assert len(response.families) == 2
+    assert response.total_family_hits == 3
+
+    # Family increment
+    request = SearchParameters(
+        query_string=query_string,
+        limit=limit,
+        max_hits_per_family=max_hits_per_family,
+        continuation_tokens=[family_continuation],
+    )
+    response = vespa_search(fake_vespa_credentials, request)
+    assert len(response.families) == 1
+    assert response.total_family_hits == 3
+
+    # Family should have changed
+    second_family_ids = [f.id for f in response.families]
+    assert sorted(first_family_ids) != sorted(second_family_ids)
+    # As this is the end of the results we also expect no more tokens
+    assert response.continuation_token is None
+
+
+@pytest.mark.vespa
+def test_vespa_search_adaptor__continuation_tokens__passages(fake_vespa_credentials):
+    query_string = "the"
+    limit = 1
+    max_hits_per_family = 10
+
+    # Make an initial request to get continuation tokens and results for comparison
+    request = SearchParameters(
+        query_string=query_string,
+        limit=limit,
+        max_hits_per_family=max_hits_per_family,
+    )
+    initial_response = vespa_search(fake_vespa_credentials, request)
+
+    # Collect family & hits for comparison later
+    initial_family_id = initial_response.families[0].id
+    initial_passages = [h.text_block_id for h in initial_response.families[0].hits]
+
+    this_continuation = initial_response.this_continuation_token
+    passage_continuation = initial_response.families[0].continuation_token
+
+    # Passage Increment
+    request = SearchParameters(
+        query_string=query_string,
+        limit=limit,
+        max_hits_per_family=max_hits_per_family,
+        continuation_tokens=[this_continuation, passage_continuation],
```

Comment:
Out of interest if these are just a list of strings how does vespa know which one is in reference to the families or passages? Is it just a clever format that vespa can understand? 

Just thinking if the order of the tokens changed would this still work?

Reply:
Example token at the family level: `BGAAABGBCBC`
And another at the passage level: `BKAAAAABKCBEBC`

So a family and a passage can go together in any order, (maybe they use the length of the string to determine it — I don't actually know).

As a side note, the ordering does become important if both tokens are from the same level of grouping. The docs say its only the [last one thats used](https://docs.vespa.ai/en/reference/grouping-syntax.html#:~:text=This%20is%20because%20continuations%20are%20processed%20in%20the%20order%20given%2C%20and%20they%20replace%20whatever%20continuations%20they%20collide%20with.)